cloud/scope/vpc_cluster.go

```diff
@@ -1646,3 +1711,340 @@ func (s *VPCClusterScope) createSecurityGroupRuleRemote(remote infrav1beta2.VPCS
+				return nil, fmt.Errorf("error parsing load balancer subnet, no id or name provided: %s", loadBalancer.Name)
+			}
+		}
+	} else {
+		var err error
+		subnetIDs, err = s.GetControlPlaneSubnetIDs()
+		if err != nil {
+			return nil, fmt.Errorf("error collecting subnet IDs for load balancer creation: %w", err)
+		}
+	}
+	return subnetIDs, nil
+}
+
+// getLoadBalancerSecurityGroupIDs will collect the ID's of the desired Security Groups for a Load Balancer.
+func (s *VPCClusterScope) getLoadBalancerSecurityGroupIDs(loadBalancer infrav1beta2.VPCLoadBalancerSpec) ([]string, error) {
+	securityGroupIDs := make([]string, 0)
+	// If SecurityGroups were provided for the load balancer, find ID's, if necessary, and use them.
+	if loadBalancer.SecurityGroups != nil {
+		for _, securityGroup := range loadBalancer.SecurityGroups {
+			if securityGroup.ID != nil {
+				securityGroupIDs = append(securityGroupIDs, *securityGroup.ID)
+				continue
+			}
+			if securityGroup.Name != nil {
+				// A Security Group may not be managed or tracked by CAPI (an existing Security Group), so do not expect it must exist in Status.
+				securityGroupID, err := s.GetSecurityGroupID(*securityGroup.Name)
+				if err != nil {
+					return nil, fmt.Errorf("error looking up load balancer security group by name %s: %w", *securityGroup.Name, err)
+				} else if securityGroupID == nil {
+					return nil, fmt.Errorf("error load balancer security group not found: %s", *securityGroup.Name)
+				}
+				securityGroupIDs = append(securityGroupIDs, *securityGroupID)
+			} else {
+				return nil, fmt.Errorf("error parsing load balancer security group, no id or name provided: %s", loadBalancer.Name)
+			}
+		}
+	}
+	return securityGroupIDs, nil
+}
+
+// buildLoadBalancerBackendPool will build a Load Balancer Pool based on the provided spec.
+func (s *VPCClusterScope) buildLoadBalancerBackendPool(pool infrav1beta2.VPCLoadBalancerBackendPoolSpec) vpcv1.LoadBalancerPoolPrototype {
+	monitor := &vpcv1.LoadBalancerPoolHealthMonitorPrototype{
+		Delay:      ptr.To(pool.HealthMonitor.Delay),
+		MaxRetries: ptr.To(pool.HealthMonitor.Retries),
+		Timeout:    ptr.To(pool.HealthMonitor.Timeout),
+		Type:       ptr.To(string(pool.HealthMonitor.Type)),
+	}
+	if pool.HealthMonitor.Port != nil {
+		monitor.Port = pool.HealthMonitor.Port
+	}
+	if pool.HealthMonitor.URLPath != nil {
+		monitor.URLPath = pool.HealthMonitor.URLPath
+	}
+	backendPool := vpcv1.LoadBalancerPoolPrototype{
+		Algorithm:     ptr.To(string(pool.Algorithm)),
+		HealthMonitor: monitor,
+		Protocol:      ptr.To(string(pool.Protocol)),
+	}
+	// Only apply a name if one was provided (otherwise rely on generated name from VPC service).
+	if pool.Name != nil {
+		backendPool.Name = pool.Name
+	}
+
+	return backendPool
+}
+
+// getDefaultBalancerBackendPools returns a list of default Load Balancer Backend Pools for a Load Balancer.
+func (s *VPCClusterScope) getDefaultLoadBalancerBackendPools() []vpcv1.LoadBalancerPoolPrototype {
```

Comment:
Any reason for creating a slice and appending one item and in caller again unpacking the slice?
Can't we return the item directly?

Reply:
I did this if any additional pools were necessary to add to the defaults.